federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -728,7 +728,9 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
+				if err := s.ensureDnsRecords(clusterName, service); err != nil {
+					return fmt.Errorf("Failed to update DNS records for service %v due to %v", service, err)
```

Comment:
Agreed, we can't just bail out like this, unfortunately, as that will potentially cause updates of all services and clusters to fail.  I'd be in support of logging an error here, and not incrementing the ensuredCount if the returned value from ensureDnsRecords is not nil.